src/transaction/log_replication.cpp

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "log_replication.hpp"
+
+#include "log_impl.h"
+#include "log_recovery.h"
+#include "log_recovery_redo.hpp"
+#include "recovery.h"
+#include "thread_looper.hpp"
+#include "thread_manager.hpp"
+#include "transaction_global.hpp"
+
+#include <cassert>
+#include <chrono>
+#include <functional>
+
+namespace cublog
+{
+
+  replicator::replicator (const log_lsa &start_redo_lsa)
+    : m_redo_lsa { start_redo_lsa }
+  {
+    log_zip_realloc_if_needed (m_undo_unzip, LOGAREA_SIZE);
+    log_zip_realloc_if_needed (m_redo_unzip, LOGAREA_SIZE);
+
+    // Create the daemon
+    cubthread::looper loop (std::chrono::milliseconds (1));   // loop every 1 millisecond
```

Comment:
I changed the comment. Is it better? :D

Reply:
After a discussion, it turns out that the aggregated behavior is as described by me above since the function `redo_upto_nxio_lsa` is implemented in a run-to-exhaustion mode and, thus, the daemon will only be put to sleep for 1 milli when there's nothing more to consume.